alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportGetFindingsAction.kt

```diff
@@ -122,19 +127,48 @@ class TransportGetFindingsSearchAction @Inject constructor(
         client.threadPool().threadContext.stashContext().use {
             scope.launch {
                 try {
-                    val getFindingsResponse = search(searchSourceBuilder)
+                    val indexName = resolveFindingsIndexName(getFindingsRequest)
+                    val getFindingsResponse = search(searchSourceBuilder, indexName)
                     actionListener.onResponse(getFindingsResponse)
+                } catch (t: AlertingException) {
+                    actionListener.onFailure(t)
                 } catch (t: Exception) {
                     actionListener.onFailure(AlertingException.wrap(t))
                 }
             }
         }
     }
 
-    suspend fun search(searchSourceBuilder: SearchSourceBuilder): GetFindingsResponse {
+    suspend fun resolveFindingsIndexName(findingsRequest: GetFindingsRequest): String {
```

Comment:
same as above - we should verify the name resolution logic via unit test.

Reply:
same test coverage as detailed for get alerts.